sympy/matrices/tests/test_matrices.py

```diff
@@ -2737,7 +2737,6 @@ def test_pinv_solve():
     # Proof the solution is not exact.
     assert A * A.pinv() * B != B
 
```

Comment:
Let's add an `@XFAIL` for rank deficient matrices where denationalization fails.